code/go/0chain.net/blobbercore/handler/grpchandler.go

```diff
@@ -52,6 +57,14 @@ func unaryDatabaseTransactionInjector() grpc.UnaryServerInterceptor {
 	}
 }
 
+func unaryTimeoutInterceptor() grpc.UnaryServerInterceptor {
+	return func(ctx context.Context, req interface{}, info *grpc.UnaryServerInfo, handler grpc.UnaryHandler) (interface{}, error) {
+		deadline := time.Now().Add(time.Duration(TIMEOUT_SECONDS * time.Second))
+		ctx, _ = context.WithDeadline(ctx, deadline)
+		return handler(ctx, req)
```

Comment:
This should return request timeout error when deadline has exceeded, by checking the done channel.

Reply:
@shravanshetty1 
it does. it already does.

context lib and grpc lib do that together, no custom code to create and pass error is needed there.
context checks for deadline and send cancel signal up, grpc transforms the signal to specific error and send it back as one of the handler returned param. all the "outer" handlers also get the error.

specifically, handler returns and clients receive such error:
```
rpc error: code = DeadlineExceeded desc = context deadline exceeded
```

Reply:
https://grpc.io/blog/deadlines/

According to this article they do handle deadlines, but couldnt find the code for it. Will test if it works and then approve.

Reply:
@shravanshetty1 
if no other changes needed, please, mark the review accordingly. thanks.